metric_tank/mdata/aggmetric.go

```diff
@@ -150,11 +150,8 @@ func (a *AggMetric) getChunkByT0(ts uint32) *chunk.Chunk {
 }
 
 func (a *AggMetric) getChunk(pos int) *chunk.Chunk {
-	if pos < 0 {
-		return nil
-	}
-	if pos >= len(a.Chunks) {
-		return nil
+	if pos < 0 || pos >= len(a.Chunks) {
+		panic(fmt.Sprintf("aggmetric %s queried for chunk %d out of %d chunks", a.Key, pos, len(a.Chunks)))
```

Comment:
Should we panic here, though? I'd like to handle it more gracefully if at all possible, rather than still panic earlier in the process than before and bringing metric_tank down.